keps/sig-node/4033-group-driver-detection-over-cri/README.md

```diff
@@ -624,307 +324,132 @@ CgroupDriver as they must be today.
 
 ###### What specific metrics should inform a rollback?
 
-<!--
-What signals should users be paying attention to when the feature is young
-that might indicate a serious problem?
--->
-
 Nodes being in NotReady state with kubelet logs indicating an error in the
 RuntimeConfig CRI request, making kubelet fail to start.
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
-<!--
-Describe manual testing that was done and the outcomes.
-Longer term, we may want to require automated upgrade/rollback tests, but we
-are missing a bunch of machinery and tooling and can't do that now.
--->
-
 Not planned as there is no persistent state associated with the feature. Manual
 testing of the feature gate (in addition to the unit tests) is performed.
 
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
 
-<!--
-Even if applying deprecation policies, they may still surprise some users.
--->
+Yes.
 
-Yes, the CgroupDriver field of the Kubelet configuration (and the corresponding
-`--cgroup-driver` flag) will be marked as deprecated, but won’t be dropped
-until wide adoption of the field on CRI has been done.
+In alpha and beta, the CgroupDriver field of the Kubelet configuration (and the
+corresponding `--cgroup-driver` flag) will be marked as deprecated.
 
-### Monitoring Requirements
-
-<!--
-This section must be completed when targeting beta to a release.
+In GA, the CgroupDriver configuration option and the `--cgroup-driver` flag are
+deprecated, will have no effect, and will be removed in a future release as per
+the Kubernetes deprecation policy.
 
-For GA, this section is required: approvers should be able to confirm the
-previous answers based on experience in the field.
--->
+### Monitoring Requirements
 
 ###### How can an operator determine if the feature is in use by workloads?
 
-<!--
-Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
-checking if there are objects with field X set) may be a last resort. Avoid
-logs or events for this purpose.
--->
-
 Kubelet and container runtime version. The
```

Comment:
if my nodes are going NotReady, how can I tell that it is due to this feature, without having to log in to all 10,000 nodes?

Reply:
This should not now happen as long as the fallback behavior is in place.

After the deprecated functionality is removed this will be similar to other kubelet startup errors. The node will be set to NotReady by the `node-controller` (visible in node events). When the kubelet does not start I don't know what other methods you have other than examine kubelet logs (log in or via scraped logs)

Thoughts @haircommander ?

Reply:
could maybe have an event emitted but that's not all that reliable